dartsim/src/EntityManagementFeatures.cc

```diff
@@ -393,22 +400,19 @@ const std::string &EntityManagementFeatures::GetLinkName(
 std::size_t EntityManagementFeatures::GetLinkIndex(
     const Identity &_linkID) const
 {
-  return this->ReferenceInterface<LinkInfo>(_linkID)
-      ->link->getIndexInSkeleton();
+  return this->links.idToIndexInContainer.at(_linkID);
 }
 
 /////////////////////////////////////////////////
 Identity EntityManagementFeatures::GetModelOfLink(
     const Identity &_linkID) const
 {
-  const DartSkeletonPtr &model =
-      this->ReferenceInterface<LinkInfo>(_linkID)->link->getSkeleton();
+  const std::size_t modelID = this->links.idToContainerID.at(_linkID);
 
   // If the model containing the link doesn't exist in "models", it means this
   // link belongs to a removed model.
-  if (this->models.HasEntity(model))
+  if (this->models.HasEntity(modelID))
   {
-    const std::size_t modelID = this->models.IdentityOf(model);
     return this->GenerateIdentity(modelID, this->models.at(modelID));
   }
   else
```

Comment:
If the model has been removed, should we remove the model from `this->models` before returning `this->GenerateInvalidId()`?

Reply:
Similar to my previous [comment](https://github.com/ignitionrobotics/ign-physics/pull/227#discussion_r598841185), I don't think we should have side effects in this function.